src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/scatterPlot.js

```diff
@@ -1,171 +1,560 @@
         }
 
-        let width     = this.dom.getAttributeNS(null, "width");
-        let height    = this.dom.getAttributeNS(null, "height");
-        const divElem = this.createDivElem(width, height);
+        let width        = this.dom.getAttributeNS(null,"width");
+        let height       = this.dom.getAttributeNS(null,"height");
+        const btnPadding = 25;
+        height           = height - btnPadding;
+        const divElem    = this.createDivElem(width,height);
+        this.dom.appendChild(divElem);
+
+        let parsedData = JSON.parse(data);
+        let axis       = parsedData.axis || {x:{scale:"linear"},y:{scale:"linear"}};
+        let focus      = parsedData.focus;
+        let points     = parsedData.points || {labels:"invisible"};
+        let dataPoints = parsedData.data || {};
 
-        let margin = {top: 20, right: 20, bottom: 20, left: 20};
-        width      = width - margin.left - margin.right;
-        height     = height - margin.top - margin.bottom;
+        let margin     = this.getMargins(axis);
+        let box_width  = width - margin.left - margin.right;
+        let box_height = height - margin.top - margin.bottom;
 
         let svg = d3.select(divElem)
             .append("svg")
-            .attr("width", width + margin.left + margin.right)
-            .attr("height", height + margin.top + margin.bottom)
+            .attr("width",width)
+            .attr("height",height)
             .append("g")
-            .attr("transform", "translate(" + margin.left + "," + margin.top + ")");
-
-        // FIXME : SVG eagerly gets all pointer events from top of it, even if
-        //         node overlaps it. Should be debugged with (#797).
-
-        let parsedData  = JSON.parse(data);
-        let dataSource  = parsedData.source || "";
-        let colorDomain = []
-        let colorRange  = []
-        if (parsedData.colors !== undefined) {
-            parsedData.colors.forEach(d => {
-                colorDomain.push(d.domain);
-                colorRange.push("#" + d.color);
-            });
+            .attr("transform","translate(" + margin.left + "," + margin.top + ")");
+
+        let extremesAndDeltas = this.getExtremesAndDeltas(dataPoints);
+
+        let scaleAndAxis = this.createAxes(axis,extremesAndDeltas,box_width,box_height,svg,focus);
+
+        this.createLabels(axis,svg,box_width,margin,box_height);
+
+        let scatter = this.createScatter(svg,box_width,box_height,points,dataPoints,scaleAndAxis);
+
+        let zoom = this.addPanAndZoom(box_width,box_height,svg,margin,scaleAndAxis,scatter,points);
+
+        // TODO: Visualization selector obfuscates button, so it is now on the bottom, should be on top.
+        this.createButtonFitAll(scaleAndAxis,scatter,points,extremesAndDeltas,zoom,box_width);
+
+        let selectedZoomBtn = this.createButtonScaleToPoints();
+
+        this.addBrushing(box_width,box_height,scatter,scaleAndAxis,selectedZoomBtn,points,zoom);
```

Comment:
I had here comments in my previous pr about newlines - they are not applied nor discussion is done here